src/js/binaryen.js-post.js

```diff
@@ -2505,36 +2524,48 @@ function wrapModule(module, self = {}) {
   };
   self['getMemoryInfo'] = function() {
     var memoryInfo = {
-      'module': UTF8ToString(Module['_BinaryenMemoryImportGetModule'](module)),
-      'base': UTF8ToString(Module['_BinaryenMemoryImportGetBase'](module)),
-      'initial': Module['_BinaryenMemoryGetInitial'](module),
-      'shared': Boolean(Module['_BinaryenMemoryIsShared'](module))
+      'module': self['memory']['module'](),
+      'base': self['memory']['base'](),
+      'initial': self['memory']['initial'](),
+      'shared': self['memory']['shared']()
     };
-    if (Module['_BinaryenMemoryHasMax'](module)) {
-      memoryInfo['max'] = Module['_BinaryenMemoryGetMax'](module);
+    var max = self['memory']['max']();
+    if (max) {
+      memoryInfo['max'] = max;
     }
     return memoryInfo;
   };
   self['getNumMemorySegments'] = function() {
     return Module['_BinaryenGetNumMemorySegments'](module);
   };
+
+  self['memorySegment'] = {
+    'offset'(id) {
+      return Module['_BinaryenGetMemorySegmentByteOffset'](module, id);
+    },
+    'data'(id) {
+      const size = Module['_BinaryenGetMemorySegmentByteLength'](module, id);
+      const ptr = _malloc(size);
+      Module['_BinaryenCopyMemorySegmentData'](module, id, ptr);
+      const res = new Uint8Array(size);
+      res.set(new Uint8Array(buffer, ptr, size));
+      _free(ptr);
+      return res.buffer;
+    },
+    'passive'(id) {
+      return Boolean(Module['_BinaryenGetMemorySegmentPassive'](module, id));
+    }
+  }
```

Comment:
Another suggestion to make this more uniform with other code. Move / rename these to

```
Module#getMemorySegmentByteOffset
Module#getMemorySegmentData
Module#isMemorySegmentPassive
```

Then becomes

```js
myModule.getMemorySegmentByteOffset(segmentIndex)
```

instead of

```js
myModule.memorySegment.offset(id)
```

Apart from adhering to surrounding code style, this also has the advantage that these won't conflict with eventual setters.